compiler/src/dotty/tools/dotc/typer/TyperPhase.scala

```diff
@@ -103,19 +75,26 @@ class FrontEnd extends Phase {
   override def runOn(units: List[CompilationUnit])(using Context): List[CompilationUnit] =
     val unitContexts =
       for unit <- units yield
-        report.inform(s"compiling ${unit.source}")
-        ctx.fresh.setCompilationUnit(unit).withRootImports
-    unitContexts.foreach(parse(using _))
-    record("parsedTrees", ast.Trees.ntrees)
+        val newCtx = ctx.fresh.setPhase(this.start).setCompilationUnit(unit)
+        // TODO without this test, dotty.tools.repl.ScriptedTests fails. Not sure why.
+        if (this.start > Periods.FirstPhaseId)
+          newCtx.withRootImports
+        else
+          newCtx
```

Comment:
🤔 probably not a big deal, but would be nice to lift this into some REPL or REPL/test code.

I assume it has something to do with how the REPL splits parsing and typing.

Reply:
Happy to take suggestions on how to fix this, but it's behind my limited understanding of how the REPL and the compiler interact.